yearn/contracts/SaddleStrategy.sol

```diff
@@ -0,0 +1,396 @@
+}
+
+/// @notice Interface for the Uniswap v2 router.
+/// @dev This is an interface with just a few function signatures of the
+///      router contract. For more info and function description please see:
+///      https://uniswap.org/docs/v2/smart-contracts/router02.
+interface IUniswapV2Router {
+    function swapExactTokensForTokens(
+        uint256,
+        uint256,
+        address[] calldata,
+        address,
+        uint256
+    ) external;
+}
+
+/// @title SaddleStrategy
+/// @notice This strategy is meant to be used with the Saddle tBTC v2 pool vault.
+///         The vault's underlying token (a.k.a. want token) should be the LP
+///         token of the tBTC v2 Saddle pool. This strategy borrows the vault's
+///         underlying token up to the debt limit configured for this strategy
+///         in the vault. In order to make the profit, the strategy deposits
+///         the borrowed tokens into the tBTC v2 Saddle reward pool. Depositing
+///         tokens in the reward pool generates KEEP token rewards.
+///         The financial outcome is settled upon a call of the `harvest` method
+///         (BaseStrategy.sol). Once that call is made, the strategy gets the
+///         KEEP token rewards from Saddle reward pool. These reward tokens are
+///         then used to buy wBTC via a decentralized exchange. At the end, the
+///         strategy takes acquired wBTC and deposits them to the Saddle tBTC v2
+///         pool. This way it obtains new LP tokens the vault is interested in,
+///         and makes the profit in result. At this stage, the strategy may
+///         repay some debt back to the vault, if needed. The entire cycle
+///         repeats for the strategy lifetime, so all gains are constantly
+///         reinvested. Worth to flag that the current implementation uses wBTC
+///         as the intermediary token because of its liquidity and ubiquity in
+///         BTC-based Curve pools.
+/// @dev Implementation is based on:
+///      - General Yearn strategy template
+///        https://github.com/yearn/brownie-strategy-mix
+contract SaddleStrategy is BaseStrategy {
+    using SafeERC20 for IERC20;
+    using Address for address;
+    using SafeMath for uint256;
+
+    // Address of the KEEP token contract.
+    address public constant keep = 0x85Eee30c52B0b379b046Fb0F85F4f3Dc3009aFEC;
+    // Address of the WETH token contract.
+    address public constant weth = 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2;
+    // Address of the WBTC token contract.
+    address public constant wbtc = 0x2260FAC5E5542a773Aa44fBCfeDf7C193bc2C599;
+    // Address of the Uniswap V2 router contract.
+    address public constant uniswap =
+        0x7a250d5630B4cF539739dF2C5dAcb4c659F2488D;
+
+    // Address of the tBTC v2 Saddle pool swap contract, see
+    // https://docs.saddle.finance/contracts
+    address public tbtcSaddlePoolSwap;
+    // Address of the tBTC v2 Saddle reward pool.
+    address public tbtcSaddleLPRewards;
```

Comment:
Can we add 'Pool' at the end, ie `tbtcSaddleLPRewardsPool`?

Reply:
Maybe leave it as is and just modify the comment to `Address of the tBTC v2 Saddle LP rewards contract`? I'm afraid adding the `pool` suffix may cause confusion with the Saddle pool.

Reply:
Ok, let's leave it as-is and just change a comment if you agree with that Tomasz.

Reply:
I agree with that and changed the comment in https://github.com/keep-network/tbtc-v2/pull/13/commits/802c155d8e6b78bfe17a77572814367052f3b4f7.